src/main/java/org/folio/rest/transaction/AllOrNothingHandler.java

```diff
@@ -0,0 +1,202 @@
+
+import org.folio.rest.jaxrs.model.Transaction;
+import org.folio.rest.jaxrs.resource.FinanceStorageTransactions;
+import org.folio.rest.persist.HelperUtils;
+import org.folio.rest.persist.Tx;
+import org.folio.rest.persist.Criteria.Criterion;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.handler.impl.HttpStatusException;
+
+public abstract class AllOrNothingHandler extends AbstractTransactionHandler {
+
+  private String temporaryTransactionTable;
+  private String summaryTable;
+  private boolean isLastRecord;
+
+  AllOrNothingHandler(String temporaryTransactionTable, String summaryTable, Map<String, String> okapiHeaders, Context context,
+      Handler<AsyncResult<Response>> handler) {
+    super(okapiHeaders, context, handler);
+    this.temporaryTransactionTable = temporaryTransactionTable;
+    this.summaryTable = summaryTable;
+    this.isLastRecord = false;
+  }
+
+  public void createTransaction(Transaction transaction) {
+    processAllOrNothing(transaction, this::createPermanentTransactions).setHandler(result -> {
+      if (result.failed()) {
+        HttpStatusException cause = (HttpStatusException) result.cause();
+        if (cause.getStatusCode() == Response.Status.BAD_REQUEST.getStatusCode()) {
+          getAsyncResultHandler().handle(Future.succeededFuture(
+              FinanceStorageTransactions.PostFinanceStorageTransactionsResponse.respond400WithTextPlain(cause.getPayload())));
+        } else {
+          getAsyncResultHandler().handle(Future.succeededFuture(FinanceStorageTransactions.PostFinanceStorageTransactionsResponse
+            .respond500WithTextPlain(Response.Status.INTERNAL_SERVER_ERROR.getReasonPhrase())));
+        }
+      } else {
+        log.debug("Preparing response to client");
+        getAsyncResultHandler().handle(
+            Future.succeededFuture(FinanceStorageTransactions.PostFinanceStorageTransactionsResponse.respond201WithApplicationJson(
+                transaction, FinanceStorageTransactions.PostFinanceStorageTransactionsResponse.headersFor201()
+                  .withLocation(TRANSACTION_LOCATION_PREFIX + transaction.getId()))));
+      }
+    });
+  }
+
+  abstract String getId(Transaction transaction);
+
+  abstract Criterion getCriterion(String value);
+
+  private Future<Tx<List<Transaction>>> createPermanentTransactions(Tx<List<Transaction>> tx) {
+    Promise<Tx<List<Transaction>>> promise = Promise.promise();
+    List<Transaction> transactions = tx.getEntity();
+    tx.getPgClient()
+      .saveBatch(tx.getConnection(), TRANSACTION_TABLE, transactions, reply -> {
+        if (reply.failed()) {
+          handleFailure(promise, reply);
+        } else {
+          promise.complete(tx);
+        }
+      });
+    return promise.future();
+  }
+
+  Future<Void> processAllOrNothing(Transaction transaction,
```

Comment:
Javadocs with some comments will be helpful to understand what this method is doing.